File: trunk/java/log_client_java/src/com/persiste/sdk/Log.java
/**
* Persiste. Java SDK
*
* The cloud based logging platform.
*
* @package             Persiste. SDK
* @author              Evance Soumaoro
* @copyright           Copyright (c) 2012 - 20xx, Evansofts.
* @license             Evansofts Proprietary Licence (EPL)
* @link                http://evansofts.com
* @version             Version 0.1
*/

package com.persiste.sdk;

import java.util.HashMap;
import java.util.Map;

public class Log {
    public int level;
    public String title;
    public String description;
    public String emiter;
    public Map<String, String> custom_fields;
    public String stack_trace;
    
    public Log(int level, String title, String description,String emiter,Map<String, String> custom_fields, String stack_trace){
    	this.level= level;
        this.title= title;
         this.description= description;
         this.stack_trace= stack_trace;
         this.custom_fields= custom_fields;
         this.emiter= emiter;
    }
    
    public Log(int level, String title, String description,String emiter,Map<String, String> custom_fields){
    	 this.level= level;
         this.title= title;
         this.description= description;
         this.stack_trace= "";
         this.custom_fields= custom_fields;
         this.emiter= emiter;
    }
    
    public Log(int level, String title, String description,String emiter){
   	 this.level= level;
        this.title= title;
        this.description= description;
        this.stack_trace= "";
        this.custom_fields= new HashMap<String, String>();
        this.emiter= emiter;
    }  
    
    public Log(int level, String title, String description){
    	this.level= level;
    	this.title= title;
    	this.description= description;
    	this.stack_trace= "";
    	this.custom_fields= new HashMap<String, String>();
    	this.emiter= "";
    }  


}

File: csharp/log_client_cs/PersisteDotSdk/Log.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PersisteDotSdk {
    public class Log {
        public int level;
        public String title;
        public String description;
        public String emiter;
        public Dictionary<String, String> custom_fields;
        public String stack_trace;

        public Log(int level, String title, String description, String emiter, Dictionary<String, String> custom_fields, String stack_trace) {
            this.level = level;
            this.title = title;
            this.description = description;
            this.stack_trace = stack_trace;
            this.custom_fields = custom_fields;
            this.emiter = emiter;
        }

        public Log(int level, String title, String description, String emiter, Dictionary<String, String> custom_fields) {
            this.level = level;
            this.title = title;
            this.description = description;
            this.stack_trace = "";
            this.custom_fields = custom_fields;
            this.emiter = emiter;
        }

        public Log(int level, String title, String description, String emiter) {
            this.level = level;
            this.title = title;
            this.description = description;
            this.stack_trace = "";
            this.custom_fields = new Dictionary<String, String>();
            this.emiter = emiter;
        }

        public Log(int level, String title, String description) {
            this.level = level;
            this.title = title;
            this.description = description;
            this.stack_trace = "";
            this.custom_fields = new Dictionary<String, String>();
            this.emiter = "";
        }


    }
}
